overlord/hookstate/hookmgr.go

```diff
@@ -423,6 +429,17 @@ func (m *HookManager) runHook(context *Context, snapst *snapstate.SnapState, hoo
 		m.contextsMutex.Unlock()
 	}()
 
+	if ph, ok := context.Handler().(Precondition); ok {
```

Comment:
It would be a good idea to have a small test that covers the case where the handler doesn't implement the `Precondition` interface.

Reply:
That's a fair point and I tried but didn't think it was worth it. I tried adding a separate suite so we'd a handler that doesn't implement it but the setup is very involved and there's nothing to test if we don't implement the interface (only that a handler w/o it would still work which we already test because we have those). So I opted not to add that